From 809612f096558cb3e7cbe23c79b5e1954fcc34f9 Mon Sep 17 00:00:00 2001
From: Learath <learath2@gmail.com>
Date: Wed, 6 May 2020 13:43:30 +0300
Subject: [PATCH] 1844: Use the virtual size to determine the section a RVA
 belongs to in PE files

Use the virtual size to determine section
---
 .../src/main/java/ghidra/app/util/bin/format/pe/NTHeader.java | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/NTHeader.java b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/NTHeader.java
index b7fa6cfd5..5ae65006b 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/NTHeader.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/app/util/bin/format/pe/NTHeader.java
@@ -145,7 +145,7 @@ public class NTHeader implements StructConverter, OffsetValidator {
 		SectionHeader[] sections = fileHeader.getSectionHeaders();
 		for (SectionHeader section : sections) {
 			long sectionVA = Integer.toUnsignedLong(section.getVirtualAddress());
-			long rawSize = Integer.toUnsignedLong(section.getSizeOfRawData());
+			long vSize = Integer.toUnsignedLong(section.getVirtualSize());
 			long rawPtr = Integer.toUnsignedLong(section.getPointerToRawData());
 
 			switch (layout) {
@@ -153,7 +153,7 @@ public class NTHeader implements StructConverter, OffsetValidator {
 					return rva;
 				case FILE:
 				default:
-					if (rva >= sectionVA && rva < sectionVA + rawSize) {
+					if (rva >= sectionVA && rva < sectionVA + vSize) {
 						return rva + rawPtr - sectionVA;
 					}
 					break;
-- 
2.38.1

